lib/contents.php

```diff
@@ -142,30 +52,35 @@ function getContents(
     }
 
     $cache = RssBridge::getCache();
-    $cache->setScope('server');
-    $cache->setKey([$url]);
-
-    if (!Debug::isEnabled() && $cache->getTime() && $cache->loadData(86400 * 7)) {
-        $config['if_not_modified_since'] = $cache->getTime();
+    $cacheKey = 'server_' . $url;
+
+    /** @var Response $cachedResponse */
+    $cachedResponse = $cache->get($cacheKey);
+    if ($cachedResponse) {
+        // considering popping
+        $cachedLastModified = $cachedResponse->getHeader('last-modified');
+        if ($cachedLastModified) {
+            $cachedLastModified = new \DateTimeImmutable($cachedLastModified);
+            $config['if_not_modified_since'] = $cachedLastModified->getTimestamp();
+        }
     }
 
     $response = $httpClient->request($url, $config);
```

Comment:
If I understand right, here, if there is no "last-modified" header in the cache (and it will be often the case, as most website does not set a last-modified header for dynamic content), we will do a request, even if there is something that is not expired in the cache ?

Or I did not understand something ? :)

Reply:
yes. we do a new http request regardless whether there is a cached response. 

this mimics the previous behavior

Reply:
OK !